diff_drive_controller/src/diff_drive_controller.cpp

```diff
@@ -39,93 +38,110 @@ constexpr auto DEFAULT_TRANSFORM_TOPIC = "/tf";
+  try {
+    auto node = get_node();
+    // with the lifecycle node being initialized, we can declare parameters
+    node->declare_parameter<std::vector<std::string>>("left_wheel_names", left_wheel_names_);
+    node->declare_parameter<std::vector<std::string>>("right_wheel_names", right_wheel_names_);
+
+    node->declare_parameter<double>("wheel_separation", wheel_params_.separation);
+    node->declare_parameter<int>("wheels_per_side", wheel_params_.wheels_per_side);
+    node->declare_parameter<double>("wheel_radius", wheel_params_.radius);
+    node->declare_parameter<double>(
+      "wheel_separation_multiplier",
+      wheel_params_.separation_multiplier);
+    node->declare_parameter<double>(
+      "left_wheel_radius_multiplier",
+      wheel_params_.left_radius_multiplier);
+    node->declare_parameter<double>(
+      "right_wheel_radius_multiplier",
+      wheel_params_.right_radius_multiplier);
+
+    node->declare_parameter<std::string>("odom_frame_id", odom_params_.odom_frame_id);
+    node->declare_parameter<std::string>("base_frame_id", odom_params_.base_frame_id);
+    node->declare_parameter<std::vector<double>>("pose_covariance_diagonal", {});
+    node->declare_parameter<std::vector<double>>("twist_covariance_diagonal", {});
+    node->declare_parameter<bool>("open_loop", odom_params_.open_loop);
+    node->declare_parameter<bool>("enable_odom_tf", odom_params_.enable_odom_tf);
+
+    node->declare_parameter<int>("cmd_vel_timeout", cmd_vel_timeout_.count());
+    node->declare_parameter<bool>("publish_limited_velocity", publish_limited_velocity_);
+    node->declare_parameter<int>("velocity_rolling_window_size", 10);
+
+    node->declare_parameter<bool>("linear.x.has_velocity_limits", false);
+    node->declare_parameter<bool>("linear.x.has_acceleration_limits", false);
+    node->declare_parameter<bool>("linear.x.has_jerk_limits", false);
+    node->declare_parameter<double>("linear.x.max_velocity", 0.0);
+    node->declare_parameter<double>("linear.x.min_velocity", 0.0);
+    node->declare_parameter<double>("linear.x.max_acceleration", 0.0);
+    node->declare_parameter<double>("linear.x.min_acceleration", 0.0);
+    node->declare_parameter<double>("linear.x.max_jerk", 0.0);
+    node->declare_parameter<double>("linear.x.min_jerk", 0.0);
+
+    node->declare_parameter<bool>("angular.z.has_velocity_limits", false);
+    node->declare_parameter<bool>("angular.z.has_acceleration_limits", false);
+    node->declare_parameter<bool>("angular.z.has_jerk_limits", false);
+    node->declare_parameter<double>("angular.z.max_velocity", 0.0);
+    node->declare_parameter<double>("angular.z.min_velocity", 0.0);
+    node->declare_parameter<double>("angular.z.max_acceleration", 0.0);
+    node->declare_parameter<double>("angular.z.min_acceleration", 0.0);
+    node->declare_parameter<double>("angular.z.max_jerk", 0.0);
+    node->declare_parameter<double>("angular.z.min_jerk", 0.0);
+  } catch (...) {
```

Comment:
@bmagyar Maybe catch `std::runtime_error` and print the error? Otherwise this requires the debugger to figure out what happened.

Reply:
good insight sir! added!